extension/jaegerremotesampling/internal/http_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/jaegertracing/jaeger/thrift-gen/baggage"
+	"github.com/jaegertracing/jaeger/thrift-gen/sampling"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/confighttp"
+)
+
+func TestMissingClientConfigManager(t *testing.T) {
+	// test
+	s, err := NewHTTP(componenttest.NewNopTelemetrySettings(), confighttp.HTTPServerSettings{}, nil)
+
+	// verify
+	assert.Equal(t, errMissingClientConfigManager, err)
+	assert.Nil(t, s)
+}
+
+func TestStartAndStop(t *testing.T) {
+	// prepare
+	s, err := NewHTTP(componenttest.NewNopTelemetrySettings(), confighttp.HTTPServerSettings{}, NewClientConfigManager())
+	require.NoError(t, err)
+	require.NotNil(t, s)
+
+	// test
+	t.Run("start", func(t *testing.T) {
+		assert.NoError(t, s.Start(context.Background(), componenttest.NewNopHost()))
+	})
+
+	t.Run("shutdown", func(t *testing.T) {
+		assert.NoError(t, s.Shutdown(context.Background()))
+	})
+}
```

Comment:
I don't quite understand why they are new instances. They are both pointers to the same instance, aren't they? I didn't catch the deadlock because of a different bug (explained in an earlier comment).

Reply:
I moved the assertions out of the t.Run's, but I would still like to understand the point about different instances being used in each block.